src/pages/[locale]/[...path]/_index.ts

```diff
@@ -0,0 +1,32 @@
+import { datocmsCollection } from '@lib/datocms';
+import { type PageRouteForPath, getPagePath } from '@lib/routing/page';
+
+export async function getStaticPaths() {
```

Comment:
Could keep this in `index.astro` and try to import it in `index.md.astro`. I like this separate file as it makes it more clear it can be used across routes. And it also obfuscated the content of `index.astro` a bit. So I'm happy with this.